jetty-websocket/javax-websocket-server/src/main/java/org/eclipse/jetty/websocket/javax/server/JavaxWebSocketServerContainer.java

```diff
@@ -283,20 +311,21 @@ public long getDefaultAsyncSendTimeout()
     public int getDefaultMaxBinaryMessageBufferSize()
     {
         // TODO: warn on long -> int conversion issue
-        return (int)this._webSocketCreatorMapping.getDefaultMaxBinaryMessageSize();
+        // TODO: Should this be Filter?
```

Comment:
Careful, buffer and message size are different concepts.

Max**Type**MessageSize applies to entire message. (only applies to whole message delivery)
Max**Type**BufferSize applies to frame. (applies to whole message delivery, partial message delivery, and streaming delivery)

Reply:
But I think the Max**Type**MessageBufferSize applies to the message size.  The interface javadoc says: `Returns the default maximum size of incoming binary message that this container will buffer`
So I think we are correct to map java MessageBufferSize to core MessageSize.     Note that core doesn't really handle MessageSize, rather it just provides a "service" to let that configuration be stored on the session.